akka-contrib/docs/receive-pipeline.rst

```diff
@@ -16,33 +16,46 @@ Some Possible Use Cases
 
 Interceptors
 ------------
-So how does an interceptor look? Well, Interceptors are defined by decorator functions
-of type :class:`Receive => Receive`, where it gets the inner :class:`Receive` by parameter and
-returns a new :class:`Receive` with the decoration applied.
-Most of the times your decorators will be defined as a regular :class:`Receive` with cases
-for the messages of your interest and at some point delegate on the inner :class:`Receive`
-you get by parameter. We will talk about ignored and unhandled messages later.
+Multiple interceptors can be added to actors that mixin the :class:`ReceivePipeline` trait. 
+These interceptors can be thought of as layers: each outer interceptor advising the 
```

Comment:
Before I commit anything, how about the following which uses the decorator/delegate terminology and renames InterceptorResult to Delegation.

(Internally the code uses advice/advisor terminology - albeit privately - so will rename to line up and lessen any confusion here, too.)

Interceptors
------------
Multiple interceptors can be added to actors that mixin the `ReceivePipeline` trait. 
These interceptors internally define layers of decorators around the actor's behavior. The first interceptor
defines an outer decorator which delegates to a decorator corresponding to the second interceptor and so on, 
until the last interceptor which defines a decorator for the actor's `Receive`. 

[etc etc as before...]